July-Morning/assignment1/Assignment_1_Q1.cpp

```diff
@@ -0,0 +1,91 @@
+#include <iostream>
+#include <string>
+#include <locale> 
+#include <unordered_map>
+#include <algorithm>
+
+// Check if two strings are anagrams (case sensitive algorithm)
+// Return value:
+// true - if strings are anagrams 
+// false - if strings are not anagrams or at least one string is empty 
+bool Is_Anagram (std::string s1, std::string s2)
+{
+    if ((s1.size() == 0) || (s2.size() == 0))
+        return false;
+    if (s1.size() != s2.size())
+        return false;
+    std::unordered_map<char, int> symbol_counter;
+    for (auto& sym : s1)

```

Comment:
A string is a sequence of characters, so a foreach loop iterates on type `char`.
`char` is a primitive, so there is no need to use a reference to it.
(A reference is implemented by a pointer; which means that any further access to the character will dereference the pointer, making the code less efficient than using the primitive directly).

To use the primitive directly:
`for (char sym : s1)`
or, if you insist on using `auto`:
`for (auto sym : s1)` (without &).

There are many opinions on using `auto`. I think that for primitives it is best to use the actual type name and not hide it behind `auto`, so I am in favor or using `char` here, which will be easily identified by experienced C/C++ users.